<launch>
 <include file="$(find xbot_cruise)/launch/xml/velocity_smoother.launch.xml"/>
<!--  <include file="$(find xbot_cruise)/launch/xml/safety_controller.launch.xml"/> -->

 <arg name="odom_frame_id"   default="odom"/>
 <arg name="base_frame_id"   default="base_footprint"/>
 <arg name="global_frame_id" default="map"/>
 <arg name="odom_topic" default="odom" />
 <arg name="laser_topic" default="scan" />
 <arg name="custom_param_file" default="$(find xbot_cruise)/param/dummy.yaml"/>
 <arg name="recovery_behavior_enabled" default='false' />
 <arg name="clearing_rotation_allowed" default='false' />
 <arg name="planner_frequency" default='1.0' />

 <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

 <!--setting param.yaml-->
  <rosparam file="$(find xbot_cruise)/param/costmap_params.yaml" command="load" ns="global_costmap" />
  <rosparam file="$(find xbot_cruise)/param/costmap_params.yaml" command="load" ns="local_costmap" />

  <rosparam file="$(find xbot_cruise)/param/local_costmap.yaml" command="load" />
  <rosparam file="$(find xbot_cruise)/param/global_costmap.yaml" command="load" />

  <rosparam file="$(find xbot_cruise)/param/dwa_planner.yaml" command="load" />
  <rosparam file="$(find xbot_cruise)/param/move_base_xu.yaml" command="load" />
  <rosparam file="$(find xbot_cruise)/param/global_planner.yaml" command="load" />
  <rosparam file="$(find xbot_cruise)/param/navfn_global_planner.yaml" command="load" />


  <rosparam file="$(arg custom_param_file)" command="load" />

 <!--setting frame-->
  <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
  <param name="global_costamp/robot_base_frame" value="$(arg base_frame_id)"/>

  <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
  <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
  <param name="DWAPlannerRos/global_frame_id" value="$(arg odom_frame_id)"/>
  <param name="recovery_behavior_enabled" value="$(arg recovery_behavior_enabled)"/>
  <param name="clearing_rotation_allowed" value="$(arg clearing_rotation_allowed)"/>
  <param name="planner_frequency" value="$(arg planner_frequency)"/>


 <!--remap topics-->
  <!-- <remap from="cmd_vel" to="/cmd_vel_mux/input/empty"/> -->
  <remap from="cmd_vel" to="navigation_velocity_smoother/raw_cmd_vel"/>

  <remap from="odom" to="$(arg odom_topic)"/>
  <remap from="scan" to="$(arg laser_topic)"/>

 </node>


  <!--base controller-->


 <!-- <node pkg="xbot_cruise" type="base_controller.py" respawn="true" name="BaseController_X" output="screen">
  <remap from="/cmd_vel_mux/input/navi" to="navigation_velocity_smoother/raw_cmd_vel"/>
  <param name="~OdomTopic"  value="/xbot_position_in_map" />
  <param name="~PlanTopic"  value="/move_base/DWAPlannerROS/global_plan" />
 </node> -->
</launch>
